yesod-core/src/Yesod/Core/Content.hs

```diff
@@ -226,6 +240,53 @@ typeOctet = "application/octet-stream"
 simpleContentType :: ContentType -> ContentType
 simpleContentType = fst . B.break (== _semicolon)
 
+decoderForCharset :: Maybe B.ByteString -> L.ByteString -> TL.Text
+decoderForCharset (Just encodingSymbol)
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "utf-8")        = LE.decodeUtf8With EE.lenientDecode
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "US-ASCII")     = TL.fromStrict . fst . decodeASCIIPrefix . B.toStrict
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "latin1")       = LE.decodeLatin1
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "GB18030")      = TL.pack . Enc.decodeLazyByteString Enc.GB18030
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "windows-1251") = TL.pack . Enc.decodeLazyByteString Enc.CP1251
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "Shift_JIS")    = TL.pack . Enc.decodeLazyByteString Enc.ShiftJIS
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "Windows-31J")  = TL.pack . Enc.decodeLazyByteString Enc.CP932
+  | otherwise = LE.decodeUtf8With EE.lenientDecode
+decoderForCharset Nothing = LE.decodeUtf8With EE.lenientDecode
+
+textDecoderFor :: ContentType -> L.ByteString -> Maybe TL.Text
+textDecoderFor ct =
```

Comment:
This is point-free - we can make the below a *bit* more legible if we accept the parameter explicitly:

```suggestion
textDecoderFor :: ContentType -> L.ByteString -> Maybe TL.Text
textDecoderFor ct bytes =
```

Reply:
This makes sense, but I'll probably rename this function while I'm at. Obviously there's no behavioral difference, but `textDecoderFor` _sounds_ like a unary function that accepts a content type and returns a decoder; `decodeTextForContentType` _sounds_ like a two-place function that accepts both a content type and bytes.